hw/ip/otp_ctrl/data/otp_ctrl.hjson

```diff
@@ -0,0 +1,473 @@
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "1:0",
+          desc: '''This is the access size for for the OTP word to be read or written through
+          the direct access interface. Note that the address is always aligned to the
+          access size internally. 0 = 1 Byte, 1 = 2 Bytes, 2 = 4 Bytes, 3 = 8 Bytes.'''
+        }
+      ]
+    }
+    { multireg: { name:     "DIRECT_ACCESS_WDATA",
+                  desc:     '''Write data for direct access. Note that only 1s are effectively programmed into the array. 0s do not have any effect.
+                  ''',
+                  count:    "2", // 2 x 32bit = 64bit
+                  swaccess: "rw",
+                  hwaccess: "hro",
+                  hwqe:     "true",
+                  cname:    "WORD",
+                  fields: [
+                    { bits: "31:0"
+                    }
+                  ]
+                }
+    },
+    { multireg: { name:     "DIRECT_ACCESS_RDATA",
+                  desc:     '''Read data for direct access.
+                  ''',
+                  count:    "2", // 2 x 32bit = 64bit
+                  swaccess: "ro",
+                  hwaccess: "hwo",
+                  cname:    "WORD",
+                  fields: [
+                    { bits: "31:0"
+                      desc: ""
+                    }
+                  ]
+                }
+    },
+
+    ///////////////////////
+    // Integrity Digests //
+    ///////////////////////
+    { skipto: "0x050" }
+
+    { name: "SECRET_INTEGRITY_DIGEST_CALC",
+      desc: "Compute and program digest for secret partition.",
+      swaccess: "r0w1c",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          desc: '''Writing 1 to this register computes the partition digest and burns it into OTP.
+                Note: this effectively locks write and read access to this partition after power cycling the device.
+                '''
+        }
+      ]
+    }
+
+    { name: "HW_CFG_INTEGRITY_DIGEST_CALC",
+      desc: "Compute and program digest for the hardware config partition.",
+      swaccess: "r0w1c",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          desc: '''Writing 1 to this register computes the partition digest and burns it into OTP.
+                Note that this effectively locks write access to this partition after power cycling the device.
+                '''
+        }
+      ]
+    }
+
+    { name: "SW_CFG_INTEGRITY_DIGEST_CALC",
+      desc: "Compute and program digest for the software config partition.",
+      swaccess: "r0w1c",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          desc: '''Writing 1 to this register computes the partition digest and burns it into OTP.
+                Note that effectively locks write access to this partition after power cycling the device.
+                '''
+        }
+      ]
+    }
+
+    { name: "SECRET_INTEGRITY_DIGEST",
```

Comment:
ah i guess this really serves the the same purpose of what i mentioned above...

Reply:
yes, exactly. SW should be able to read the digests, once they are available (they are 0 by default).